templates/experiments/components/exports.html

```diff
@@ -1,30 +1,39 @@
+{% load custom_filters %}
 <div class="flex flex-row gap-1" id="chat-exports">
     <form>
         {% csrf_token %}
-        <input name="participants" class="hidden" x-bind:value="participants"/>
-        <input name="tags" class="hidden" x-bind:value="tags"/>
-        <button
-            class="btn btn-sm btn-outline btn-primary no-animation"
-            {% if task_id %}
-                disabled
-                hx-get="{% url 'experiments:get_export_download_link' team.slug experiment.id task_id %}"
-                hx-trigger="every 2s"
-            {% else %}
+        <button class="btn btn-sm btn-outline btn-primary no-animation"
                 hx-post="{% url 'experiments:generate_chat_export' team.slug experiment.id %}"
-            {% endif %}
-            hx-swap="outerHTML"
-            hx-target="#chat-exports"
-        >
+                hx-include="[name='show-all']"
+                hx-vals='js:{
+                         {% for param, value in request.GET.items %}
+                             {% if param|matches:"^filter_" %}
+                                 "{{ param }}": "{{ value|escapejs }}",
+                             {% endif %}
+                         {% endfor %}
+                         }'
```

Comment:
This will get generated at load time right? And since the button doesn't get reloaded when the filters update I don't think this is going to work correctly. If you add a filter and then export the data without refreshing the page does the filter get applied to the export?

If I'm right, maybe you could use the `HX-Current-URL` header to get the current filter set? (I haven't tested that)

Reply:
I tried `HX-Current-URL` and it gave me just the POST, but I tried this https://github.com/dimagi/open-chat-studio/pull/1321/commits/70c1ebad9a0562e9c058a4223e0a65ce8ed9e9ee

Reply:
I take that back: https://github.com/dimagi/open-chat-studio/pull/1321/commits/da757fbdc406f99bd66912d3e8a5a4d1427d1dab